colcon_cmake/task/cmake/build.py

```diff
@@ -148,25 +148,22 @@ async def _reconfigure(self, args, env):
         cmake_args += ['-DCMAKE_INSTALL_PREFIX=' + args.install_base]
         generator = get_generator(args.build_base, args.cmake_args)
         if os.name == 'nt' and generator is None:
-            vsv = get_visual_studio_version()
-            if vsv is None:
+            try:
+                msbuild_version = get_msbuild_version()
```

Comment:
The current patch re-determines the `msbuild` version for each package which seem like unnecessary overhead. Instead it should determine it only once on the first invocation and afterwards reuse the cached result.

Reply:
seems like premature optimization, but done.

Reply:
I wouldn't call invoking a process potentially hundreds of times (for larger workspaces) for no reason "premature optimization"...